datalad_fuse/fuse_.py

```diff
@@ -61,6 +63,17 @@ def destroy(self, _path=None):
             except Exception as e:
                 lgr.error("%s", e)
         self._fhdict = {}
+        cache_clear = cfg.get("datalad.fusefs.cache-clear")
+        if cache_clear == "visited":
+            for dsap in self._adapter.datasets.values():
+                dsap.clear()
+        elif cache_clear == "recursive":
+            ds = Dataset(self.root)
+            DatasetAdapter(ds.path).clear()
+            for subds in ds.subdatasets(
+                recursive=True, fulfilled=True, result_renderer="disabled"
+            ):
+                DatasetAdapter(subds["path"]).clear()
```

Comment:
I think it would be better to replace this entire block with just `ds.fsspec_cache_clear(recursive=True)`, and above with invocations without the recursive=True... or why not?

Reply:
Would invoking a Datalad extension method from within another method work?  When I was writing #33, the import of `datalad.api.Dataset` in `fsspec.py` caused the `fsspec_head()` method to not be available in the tests.

Reply:
"interesting"... I think it should work, especially since you would be just using `ds.fsspec_ccahe_clear` at run time, so it should be "bound" to Dataset by then and you don't need any extra import which might cause some circularity.

Reply:
@yarikoptic Doing `Dataset(self.root).fsspec_cache_clear(recursive=True)` works, but only if `Dataset` is imported from `datalad.distribution.dataset`; importing from `datalad.api` instead produces "AttributeError: 'Dataset' object has no attribute 'fsspec_cache_clear'".

Reply:
good to know, filed https://github.com/datalad/datalad/issues/6315 for this. Meanwhile then just import Dataset from its originating `datalad.distribution.dataset`.